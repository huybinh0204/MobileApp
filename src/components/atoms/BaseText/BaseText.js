import PropTypes from 'prop-types';
import React from 'react';
import { Text } from 'react-native';
import Animated from 'react-native-reanimated';

const BaseText = ({ children, variant, ...props }) => {
  const TextComponent = variant === 'animated' ? Animated.Text : Text;
  return <TextComponent {...props}>{children}</TextComponent>;
};

BaseText.propTypes = {
  allowFontScaling: PropTypes.bool,
  variant: PropTypes.any,
};

BaseText.defaultProps = {
  allowFontScaling: false,
  variant: 'normal',
};

export default BaseText;
