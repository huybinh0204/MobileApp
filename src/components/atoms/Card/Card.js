import styled from '@emotion/native';
import { Platform } from 'react-native';

const Card = styled.View(({ theme }) => ({
  alignSelf: 'stretch',
  backgroundColor: theme.colors.white,
  borderColor: theme.colors.beta50,
  borderRadius: 10,
  borderWidth: Platform.OS === 'android' ? 1 : 0,
  padding: theme.spacing.s,
  shadowColor: theme.colors.shadow,
  shadowOffset: {
    width: 0,
    height: 7,
  },
  shadowOpacity: 0.1,
  shadowRadius: 10,
}));

export default Card;
