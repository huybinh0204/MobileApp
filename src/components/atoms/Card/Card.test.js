import { render } from '@testing-library/react-native';
import React from 'react';
import BaseText from '_components/atoms/BaseText/BaseText';
import Theme from '_utilities/Theme';
import Card from './Card';

describe('Card', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(
      <Theme>
        <Card>
          <BaseText>test</BaseText>
        </Card>
      </Theme>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
