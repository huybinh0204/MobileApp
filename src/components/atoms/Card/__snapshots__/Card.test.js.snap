// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#F2F3F4",
        "borderRadius": 10,
        "borderWidth": 0,
        "padding": 29,
        "shadowColor": "#362E60",
        "shadowOffset": Object {
          "height": 7,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
  >
    test
  </Text>
</View>
`;
